FreeRTOS/FreeRTOS-Plus-TCP
# #556 Initial Cmake Module definition.

Support for CMake and compilation with strictest definitions for GNU/Clang.

Description
-----------
Added support for CMake compile of the source code.
- Excludes some Ports at the moment - due to lack of testing infrastructure.
- Excludes integrating/changing unit-tests and compiling:
  - `test/Coverity`
  - `test/cbmc`
  - `test/unit-tests`
  - Reason is these are projects of their own and do not know the interworkings of these - could integrate later.
- Includes build of the Various Build-configurations for testing - including a CUSTOM for externally defined configuration.
  - `DEFAULT_CONF`, `ENABLE_ALL`, `DISABLE_ALL`



Test Steps
-----------
- Updated CI - but tested locally with:
  - Ubuntu 18.04 with Clang 13 - ran all 3 build configuration tests
  - Ubuntu 18.04 with GNU 7.5 - ran all 3 build configuration tests

Related Issue
-----------
- #556 - Support for CMake.
- Note there are several Clang warnings that should be addressed in various files.  These will be fixed in subsequent PR's.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
